azkaban-web-server/src/main/java/azkaban/webapp/servlet/FlowTriggerServlet.java

```diff
@@ -75,21 +81,52 @@ private void ajaxFetchTrigger(final int projectId, final String flowId, final Se
     }
   }
 
+  private boolean checkProjectIdAndFlowId(HttpServletRequest req) {
+    return hasParam(req, "projectId") && hasParam(req, "flowId");
+  }
+
   private void handleAJAXAction(final HttpServletRequest req,
       final HttpServletResponse resp, final Session session) throws ServletException,
       IOException {
     final HashMap<String, Object> ret = new HashMap<>();
     final String ajaxName = getParam(req, "ajax");
     if (ajaxName.equals("fetchTrigger")) {
-      if (hasParam(req, "projectId") && hasParam(req, "flowId")) {
+      if (checkProjectIdAndFlowId(req)) {
         final int projectId = getIntParam(req, "projectId");
         final String flowId = getParam(req, "flowId");
         ajaxFetchTrigger(projectId, flowId, session, ret);
-        if (ret != null) {
-          this.writeJSON(resp, ret);
+      }
+    } else if (ajaxName.equals("pauseTrigger") || ajaxName.equals("resumeTrigger")) {
+      if (checkProjectIdAndFlowId(req)) {
+        final int projectId = getIntParam(req, "projectId");
+        final String flowId = getParam(req, "flowId");
```

Comment:
Please also check if "flowID" is valid or not.

Reply:
updated.